internal/engine/experiment/nwebconnectivity/control.go

```diff
@@ -0,0 +1,124 @@
+package nwebconnectivity
+
+import (
+	"context"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+
+	"github.com/ooni/probe-cli/v3/internal/engine/httpx"
+	"github.com/ooni/probe-cli/v3/internal/engine/model"
+	"github.com/ooni/probe-cli/v3/internal/errorsx"
+)
+
+// NextLocationInfo contains the redirected location as well as the request object which forwards most headers of the initial request.
+// This forwarded request is generated by the http.Client and
+type NextLocationInfo struct {
```

Comment:
Hey, isn't this structure duplicate?

Reply:
yes, and I deleted it for now, but we will later need the same thing for `nwebconnectivity` also :D